homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,688 @@
+            icon=icon,
+            unit=unit,
+            sensordata=sensordata,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the temperature sensor."""
+        _LOGGER.debug("Updating state of sensor: %s", self._name)
+
+        sensordata = None
+
+        if self._kind == "water_temperature":
+            for backyard in self.coordinator.data:
+                for bow in backyard["BOWS"]:
+                    if bow["systemId"] == self.bow["systemId"]:
+                        sensordata = bow["waterTemp"]
+                        break
+
+            self._name = (
+                self._backyard["BackyardName"]
+                + " "
+                + self.bow.get("Name")
+                + " Water Temperature"
+            )
+
+        elif self._kind == "air_temperature":
+            for backyard in self.coordinator.data:
+                if backyard.get("systemId") == self._attrs["MspSystemId"]:
+                    sensordata = backyard.get("airTemp")
+
+            self._name = self._backyard.get("BackyardName") + " Air Temperature"
+
+        temp_return = int(sensordata)
+        temp_state = int(sensordata)
+        unit_of_measurement = TEMP_FAHRENHEIT
+        if self._backyard["Unit-of-Measurement"] == "Metric":
+            temp_return = round((temp_return - 32) * 5 / 9, 1)
+            unit_of_measurement = TEMP_CELSIUS
+
+        if int(sensordata) == -1:
+            temp_return = None
+            temp_state = None
+
+        self._attrs["hayward_temperature"] = temp_return
+        self._attrs["hayward_unit_of_measure"] = unit_of_measurement
+        if temp_state is not None:
+            self._state = float(temp_state)
+            self._unit = TEMP_FAHRENHEIT
+
+        return self._state
+
+
+class OmniLogicPumpSpeedSensor(OmnilogicSensor):
+    """Define an OmniLogic Pump Speed Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensordata: dict,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            sensordata=sensordata,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the pump speed sensor."""
+        _LOGGER.debug("Updating state of sensor: %s", self._name)
+
+        sensordata = {}
+        pump_type = ""
+        pump_speed = 0
+
+        if self._kind == "filter_pump_speed":
+            for backyard in self.coordinator.data:
+                for bow in backyard.get("BOWS"):
+                    if bow["Filter"]["systemId"] == self.entitydata.get("systemId"):
```

Comment:
Store the system id for the entity as an instance attribute in the parent entity class so it's easy to access.

Reply:
Can you clarify? Not sure what the issue you are highlighting is.

Reply:
In the entity parent class `__init__` method, create an instance attribute:
```
self._system_id = self.entity_data["systemId"]
```
Then we don't need to look up the system id in the entity data when we need the system id.

Reply:
Addressed